jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -1076,6 +1173,15 @@ public Resource getBaseResource()
             return _baseResource;
         }
 
+        @Override
+        public File getTempDirectory()
+        {
+            File tempDirectory = ContextHandler.this.getTempDirectory();
```

Comment:
There's a difference between tmp directory configured at the server level and one at the context level. At  the server level, it's the _parent_ directory of all context tmp dirs, but configured at the context level, it is the directory where the temporary files are written. With the code the way it is, how would each context get it's own separate directory inside the server's directory? What am I missing?

Reply:
I don't think it is correct the say: "At the server level, it's the parent directory of all context tmp dirs".
The `WebInfConfiguration` will make context temporary dirs with the server temp directory as their parent, but that is only a convention/heutistic, otherwise:
 + A context might not have a configured temp directory, in which case the server temp directory can be directly used (if configured else `java.io.tempdir` is used).
 + a context might have a configured temp directory that is entirely unrelated to the temp director configured for the server.
 + a context might have a configured temp directory that is a child or descendant of any temp director configured for the server.

This is consistent with we do other configuration: if somebody explicitly sets some configuration, we put very few limits on what can be set, but if they don't set something then we use good conventions to come up with a reasonable value for them.

Reply:
The `WebInfConfiguration` approach is safest and ensures that multiple contexts don't overwrite each other. There is no such protection with multiple `Contexts`, but I guess as there isn't a predefined structure for on-disk info for `Contexts` like there is with `WebAppContexts` its fine to rely on caveat emptor.